eclass/coreos-go.eclass

```diff
@@ -9,41 +9,52 @@
 # @DESCRIPTION:
 # Name of the Go package unpacked to ${S}, this is required.
 
-[[ ${EAPI} != "5" ]] && die "Only EAPI=5 is supported"
+[[ ${EAPI} != "5" ]] && die "${ECLASS}: Only EAPI=5 is supported"
 
 inherit flag-o-matic multiprocessing toolchain-funcs
 
 DEPEND="dev-lang/go"
 
+# @FUNCTION: go_get_arch
+# @USAGE: export GOARCH=$(go_get_arch)
+go_get_arch() {
```

Comment:
If we wrap it with go_get_arch() then if an arch is added that doesn't match we only need to fix  go_get_arch() and not all the ebuilds that use $ARCH directly.  What do you think?